src/Panel.js

```diff
@@ -60,26 +62,32 @@ const Panel = React.createClass({
   },
 
   render() {
+    let {headerRole, panelRole, ...props} = this.props;
     return (
-      <div {...this.props}
+      <div {...props}
         className={classNames(this.props.className, this.getBsClassSet())}
         id={this.props.collapsible ? null : this.props.id} onSelect={null}>
-        {this.renderHeading()}
-        {this.props.collapsible ? this.renderCollapsibleBody() : this.renderBody()}
+        {this.renderHeading(headerRole)}
+        {this.props.collapsible ? this.renderCollapsibleBody(panelRole) : this.renderBody()}
```

Comment:
Unfortunate that we can't add `panelRole` (should this be `bodyRole`?) in the general case because of the silly stuff I added with the "fill" children. Oops.